spring-integration-core/src/main/java/org/springframework/integration/config/GlobalChannelInterceptorProcessor.java

```diff
@@ -96,6 +100,21 @@ public void afterSingletonsInstantiated() {
 				addMatchingInterceptors(entry.getValue(), entry.getKey());
 			}
 		}
+
+		this.singletonsInstantiated = true;
+	}
+
+	@Override
+	public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {
+		return bean;
+	}
```

Comment:
You don't need it here, only for backport since it uses older spring

Reply:
That's not related if we are going to back-port the fix.

It is much easier to merge and just cherry-pick without any modifications.

If there is really an interest to chase only Java 8 style, there is always a room for extra subsequent polishing, but only for the `master`.

Reply:
IMHO the interest of keeping code clean (using the tools that are currently available to you) should supersede the desire for clean cherry-pick, so it really isn't about java 8

Reply:
Gary, I see you added +1 to Oleg's comment. So, is it a new codding style I'm not aware?
We really got used to keep mergeable commits before...
Or am I missing here anything again?
Thanks

Reply:
@artembilan I was just indicating that I will remove this on master after cherry-picking to 4.3.x; that's all.

It doesn't really matter to me which way we do it, but it's probably easiest to use Java 8 on master and fix during backport (like we do with lambdas).